kuma/javascript/src/bcd.js

```diff
@@ -271,45 +271,722 @@ export function activateBCDSignals(
+            const browserLogo = document.createElement('img');
+            const browserSelectedIconWrapper = document.createElement('span');
+            const browserNameBlock = document.createElement('div');
+
+            browserBlock.className = 'browser';
+            browserSelectedIconWrapper.className = 'browser-selected-icon';
+            browserLogoWrapper.className = 'browser-logo';
+            browserNameBlock.className = 'browser-name';
+            browserNameBlock.innerText = browserName;
+
+            browserLogo.setAttribute('src', browserLogoSrc);
+            browserLogo.setAttribute('alt', browserName);
+
+            browserLogoWrapper.appendChild(browserSelectedIconWrapper);
+            browserLogoWrapper.appendChild(browserLogo);
+            browserBlock.appendChild(browserLogoWrapper);
+            browserBlock.appendChild(browserNameBlock);
+
+            browserBlock.addEventListener('click', function() {
+                this.classList.toggle('selected');
+                validateControls();
+            });
+
+            browserControlBlock.appendChild(browserBlock);
+        }
+
+        return createFormControl({
+            header: headerText,
+            description: descriptionText,
+            el: browserControlBlock,
+            index: 1
+        });
+    };
+
+    /**
+     * Builds and creates row select control with options parsed from bc table
+     * @returns Select row control as a `HTMLElement`
+     */
+    const createSelectRowControl = () => {
+        const headerText = gettext('Which table row is affected?');
+        const rowControlBlock = document.createElement('span');
+        const selectControl = document.createElement('select');
+
+        selectControl.id = 'select-row';
+        rowControlBlock.className = 'select-wrapper';
+
+        const rows = document.querySelectorAll('.bc-table th[scope=row]');
+        for (const elem of rows) {
+            const featureName = elem.innerText;
+            const option = document.createElement('option');
+            option.innerText = featureName;
+            option.setAttribute('value', featureName);
+            selectControl.appendChild(option);
+        }
+
+        rowControlBlock.appendChild(selectControl);
+        return createFormControl({
+            header: headerText,
+            description: '',
+            el: rowControlBlock,
+            index: 2,
+            inline: true
+        });
+    };
+
+    /**
+     * Builds and creates brief explanation control which is basically a textarea
+     * @returns Brief explanation control as a `HTMLElement`
+     */
+    const createBriefExplanationControl = () => {
+        const headerText = gettext('Can you provide a brief explanation?');
+        const descriptionText = gettext(
+            'Briefly outline the issue you are highlighting.'
+        );
+
+        const textAreaControl = document.createElement('textarea');
```

Comment:
For security reasons, I think we should set a `maxlength` on this, and check/enforce the length on the server-side.